### PR TITLE
Bug 1134982 - merge b2g RIL works into b2g/b2g-lollipop-mr1

### PR DESCRIPTION
**Pull request by vicamo**
This branch was created by:
1. merging https://github.com/mozilla-b2g/platform_hardware_ril/tree/b2g-ril_v7 into https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-jellybean (for RILv8)
2. and then into https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-kitkat (for RILv9)
3. and into https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-lollipop (for RILlv10)
4. and finally https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-lollipop-mr1 (for RILv11)

There is no suitable branch for merge yet, so I set pull request target to master branch.

However, there seems to be some selinux problems because I got:

```
init: Warning!  Service qemu-props needs a SELinux domain defined; please fix!
init: Warning!  Service goldfish-logcat needs a SELinux domain defined; please fix!
init: Warning!  Service goldfish-setup needs a SELinux domain defined; please fix!
init: cannot find '/system/bin/debuggerd64', disabling 'debuggerd64'
init: Warning!  Service gonksched needs a SELinux domain defined; please fix!
init: Warning!  Service fakeappops needs a SELinux domain defined; please fix!
init: Warning!  Service b2g needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy1 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy2 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy3 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy4 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy5 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy6 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy7 needs a SELinux domain defined; please fix!
init: Warning!  Service rilproxy8 needs a SELinux domain defined; please fix!
init: Warning!  Service b2g needs a SELinux domain defined; please fix!
```

so b2g is not starting up.

### CODE REVIEW

**Comment by vicamo** [comment]
I noted b2g/b2g-jellybean and b2g/b2g-kitkat are still not fast forwarded to https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-jellybean and https://github.com/vicamo/b2g_platform_hardware_ril/tree/next/b2g/b2g-kitkat ?

**Comment by EdgarChen** [comment]
I just checked the b2g/b2g-jellybean and b2g/b2g-kitkat. They are both fastforward to your branches, next/b2g/b2g-jellybean and next/b2g/b2g-kitkat.

I use following way to check,
1. Switch to b2g/b2g-jellybean
2. Set next/b2g/b2g-jellybean as upstream
3. git pull

Please let me know if I misunderstand anything.

**Comment by vicamo** [comment]
You have to do `git push` to update remote heads:

```
git remote add vicamo git@github.com:vicamo/b2g_platform_hardware_ril.git
git fetch vicamo
git push b2g vicamo/next/b2g/b2g-kitkat:refs/heads/b2g/b2g-kitkat
git push b2g vicamo/next/b2g/b2g-jellybean:refs/heads/b2g/b2g-jellybean
```

**Comment by vicamo** [comment]
by the way, you should not "merge" this pull request, just do `git push b2g vicamo/next/b2g/b2g-lollipop-mr1:refs/heads/b2g/b2g-lollipop-mr1` if you feel appropriate. This pull request will be closed automatically then.

**Comment by EdgarChen** [comment]
I did not push to remote is because I did not test locally yet, will do this soon.
And need to switch the manifest back to b2g-kitkat branch after pushing to remote.

For the lollipop, please wait a bit, we are working on fixing some booting issue.
And I will also do some test first before pushing to remote.

Thank you.

**Comment by EdgarChen** [comment]
Just update the remote branches b2g/b2g-jellybean and b2g/b2g-kitkat.
And I found there is one commit missing, https://github.com/mozilla-b2g/platform_hardware_ril/commit/1a7d1e2dcffa50a5d5be03a778848f831e0bc2b8.
I am going to cherry pick it from master to b2g-kitkat.

**Comment by vicamo** [comment]
I'll have a check again tomorrow.

**Comment by EdgarChen** [comment]
Note that https://github.com/mozilla-b2g/platform_hardware_ril/commit/1a7d1e2dcffa50a5d5be03a778848f831e0bc2b8 is only developed in kitkat (master branch), we didn't back porting it to ics (b2g-ril_v7). Thank you.

**Comment by vicamo** [comment]
Ok. Go ahead.

File: libril/ril_commands.h [hunk 1]
```diff
@@ -146,3 +146,25 @@
     {RIL_REQUEST_SHUTDOWN, dispatchVoid, responseVoid},
     {RIL_REQUEST_GET_RADIO_CAPABILITY, dispatchVoid, responseRadioCapability},
     {RIL_REQUEST_SET_RADIO_CAPABILITY, dispatchRadioCapability, responseRadioCapability},
+    /* unsupported/unused request numbers below */
+    {0, NULL, NULL}, /* 131 */
```
Comment (EdgarChen) [inline libril/ril_commands.h:150]: Unsupported/unused request starts with 132 (8b4e4f7e5ddcb8b7450bcded4d0395f02b340bbe)?

File: reference-ril/reference-ril.c [hunk 2]
```diff
@@ -3299,12 +4693,18 @@ mainLoop(void *param __unused)
...
+                }
+
                 if (!strcmp(s_device_path, "/dev/socket/qemud")) {
                     /* Before trying to connect to /dev/socket/qemud (which is
                      * now another "legacy" way of communicating with the
                      * emulator), we will try to connecto to gsm service via
                      * qemu pipe. */
-                    fd = qemu_pipe_open("qemud:gsm");
+                    char buffer[32];
+                    snprintf(buffer, sizeof(buffer), "qemud:gsm%s", s_client_id);
```
Comment (EdgarChen) [inline reference-ril/reference-ril.c:4706]: https://github.com/vicamo/b2g_platform_hardware_ril/blame/next/b2g/b2g-lollipop-mr1/rild/rild.c#L143-L145 gives a default value to `clientId` (d365219a1873e42d2ff669e13a9fe6ce6973bd51), the first rild tries to open pipe **qemu:gsm0** which makes it unable to communicate with qemu.

File: rild/rild.c [hunk 3]
```diff
@@ -211,7 +211,8 @@ int main(int argc, char **argv)
 
                 sleep(1);
 
-                fd = qemu_pipe_open("qemud:gsm");
+                snprintf(buffer, sizeof(buffer), "qemud:gsm%s", clientId);
```
Comment (EdgarChen) [inline rild/rild.c:214]: Ditto.

**Comment by EdgarChen** [comment]
Hi Vicamo, will you provide patches to fix above comment? or you wanna fix them in the merge node? Thank you.

BTW, I have ran mobileconnection marionette tests on emulator-l locally. I saw some failures (as below) and I am checking why got failure on those tests now.

```
SUMMARY
-------
 8:20.83 LOG: MainThread INFO passed: 23
 8:20.83 LOG: MainThread INFO failed: 9
 8:20.83 LOG: MainThread INFO todo: 1
 8:20.83 LOG: MainThread INFO
FAILED TESTS
-------
 8:20.83 LOG: MainThread INFO test_mobile_networks.js
 8:20.83 LOG: MainThread INFO test_mobile_operator_names.js
 8:20.83 LOG: MainThread INFO test_mobile_operator_names_plmnlist.js
 8:20.83 LOG: MainThread INFO test_mobile_operator_names_roaming.js
 8:20.83 LOG: MainThread INFO test_mobile_preferred_network_type.js
 8:20.83 LOG: MainThread INFO test_mobile_data_connection.js
 8:20.83 LOG: MainThread INFO test_mobile_data_state.js
 8:20.83 LOG: MainThread INFO test_mobile_last_known_network.js
 8:20.83 LOG: MainThread INFO test_dsds_mobile_data_connection.js
```

**Comment by EdgarChen** [comment]
I am going to make b2g-lollipop-mr1 fastforward to https://github.com/vicamo/b2g_platform_hardware_ril/commits/next/b2g/b2g-lollipop-mr1 and provide separated patches to fix above comments.

**Pull request merged**